app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+          ])
+          ->setComponents([
+            (new THead())
+              ->setComponents([
+                (new ColumnHeadersRow),
+                (new FiltersRow)
+                  ->setComponents([
+                    (new RenderFunc(function() {
+                        return "";
+                    }))
+                    ->setRenderSection('filters_row_column_name')
+                  ]),
+                (new OneCellRow)
+                  ->setRenderSection(RenderableRegistry::SECTION_END)
+                  ->setComponents([
+                    (new RecordsPerPage)
+                      ->setVariants([25, 50, 100, 1000])
+                      ->setTemplate('webapp/components/nayjest_records_per_page_template'),
+                    (new HtmlTag)
+                      ->setContent('フィルタ')
+                      ->setTagName('button')
+                      ->setRenderSection(RenderableRegistry::SECTION_END)
+                      ->setAttributes([
+                        'class' => 'btn btn-success btn-sm filter-post'
+                      ])
+                  ])
+              ])
+              ,
+            (new TFoot)
+              ->setComponents([
+                (new OneCellRow)
+                  ->setComponents([
+                    (new Pager),
+                    (new HtmlTag)
+                      ->setAttributes(['class' => 'pull-right'])
+                      ->addComponent((new ShowingRecords)->setTemplate('webapp/components/nayjest_showing_records_template')),
+                  ])
+              ])
+              ,
+          ]);
+        $grid = new Grid($cfg);
+        $grid = $grid->render();
+        return view('webapp/posts/index', [
+          'posts_grid' => $grid,
+          'action_delete' => $this->currentUser->is(User::ADMIN) ? 'delete' : 'writerDelete',
+          'is_filter' => count(array_column($request->all(), 'filters')) ? true : false
+        ]);
+    }
+
+    public function getCreate(Request $request) {
+        $categoriesSet = [];
+        $sites = Site::where('status', 1)->get();
+
+        if(count($sites) == 0) {
+            $request->session()->flash('flashMessageSet',
+              (new MessageSet())->setType('warning')->add(new Message('ターゲットサイトがありません。記事を新規作成する前にターゲットサイトを追加してください', 'warning'))
+            );
+
+            return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+        }
+
+        foreach(Category::all() as $category) {
+            if(!isset($categoriesSet[$category->site_id])) {
+                $categoriesSet[$category->site_id] = [];
+            }
+            $categoriesSet[$category->site_id][] = $category;
+        }
+
+        return view('webapp/posts/create', [
+          'sites' => $sites,
+          'categoriesSet' => $categoriesSet
+        ]);
+    }
+
+    public function postCreate(Request $request) {
+        $postTitle = $request->get('post_title');
+        $postContent = $request->get('post_content');
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "post_content" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)